editorconfig_plugin/gedit3.py

```diff
@@ -44,6 +45,5 @@ def get_document_properties(self, document):
             location = document.get_file().get_location()
             if location:
                 file_uri = location.get_uri()
-                if file_uri.startswith('file:///'):
-                    return self.get_properties_from_filename(file_uri[7:])
+                return self.get_properties_from_filename(unquote(urlparse(file_uri).path))
```

Comment:
This seems wrong to me, because what if the `file_uri` is something like `sftp://host.domain/path/to/some%20remote%20file.txt`? At least in _theory_ that's possible, with Gio... so it seems like the `if file_uri.startswith('file:///')` check is still necessary, no?

In fact, I just ran `gedit sftp://my.fileserver/home/ferd/.zshrc` and GEdit opened up my remote `.zshrc` just fine. But it'd break things, if the plugin turns that location into the path to my _local_ `.zshrc`. (Which also exists, but is not the same file!)

If the `file_uri` _does_ start with `file:///`, I agree using `urlparse` and `unquote` is a better approach to converting URLs into local pathnames.

Reply:
Alright thanks for the correction. So I guess it's better to keep the `file:///` URI checking then, let me fix this soon